.githooks/pre-commit

```diff
@@ -10,7 +10,8 @@ done
 
 if [[ -n "${STAGED_GO_FILES}" ]]; then
     set -o errexit
-    go test -race ./internal...
```

Comment:
FYI the reason I don't use make test here is that `make test` ignores cache, I wanted the pre commit to be as fast as possible so I made this change

Reply:
Ok, sounds like it'd be good to keep the cache here

Reply:
-count=1 is the flag that force tests to ignore cache, I also believe (tho I could be wrong) enabling coverage disables the cache as well